### DISCUSSION TITLE
Questions regarding configuration and sending of requests to external websites

### DISCUSSION THREAD

**Discussion by user84918342** [emad-elsaid/xlog#64] [Q&A]
First of all, great tool! I just stumbled upon this, and it looks amazing. Before, I was using mdwiki, but it hasn't been maintained for around 10 years. It doesn't look as good and isn't really extensible or developer-friendly, I would say. So, it's amazing that I found this.

I have a three general questions to get started:
- Is there a configuration file?
- Does xlog ever send requests to external websites? (For things like MathJax rendering, loading extensions, or the CSS theme.)
  - (Okay, except for link previews—of course, then it is sending requests.)
- (All the links in my current markdown files look like this: `[other page](./other/other-page.md)`. Xlog doesn't follow these links because of the `.md` extension. Can I change this behavior easily?)
  - Not as important, I will just modify it to my needs real quick
- Can I search for filename instead of text inside files?

**Comment by emad-elsaid** [comment]
Thank you for the kind words :heart: 
* Xlog doesn't depend on config files but it accepts arguments. so a shell script can work like a config file, [here is an example:](https://github.com/emad-elsaid/emad-elsaid.github.io/blob/master/serve)
* No xlog doesn't send requests to external websites all javascript and CSS are served from the binary. only the link [preview extension does that](https://github.com/emad-elsaid/xlog/blob/fbae837a34e945657568c571b188c3fa361e525c/extensions/link_preview/link_preview.go#L40)  to be sure search with `net/http` to get all files that uses this package. [here](https://github.com/search?q=repo%3Aemad-elsaid%2Fxlog%20%22net%2Fhttp%22&type=code) 
* xlog doesn't have a config/flag to switch the behavior to serve .md files. but you can do that by modifying the code. of the top of my head you can change [this line](https://github.com/emad-elsaid/xlog/blob/fbae837a34e945657568c571b188c3fa361e525c/handlers.go#L59)  to remove the suffix `.md` from the page value and it should work as you would expect serving both `.md` and without `.md` 
* the search [searches with filename](https://github.com/emad-elsaid/xlog/blob/fbae837a34e945657568c571b188c3fa361e525c/extensions/search/search.go#L53) and content of the files
